tokio/src/sync/cancellation_token.rs

```diff
@@ -232,31 +247,11 @@ impl CancellationToken {
         }
     }
 
-    /// Returns the number of child tokens
-    #[cfg(all(test, not(loom)))]
-    fn child_tokens(&self) -> usize {
```

Comment:
It seems like you also removed the assertions which made use of this method.  I would think having extra test coverage is good, isn't it?

Reply:
Hard to say... what is it testing exactly? How can it be tested using public APIs.

Reply:
How does the caller of `cancel` wait for cancellation to complete? It feels like you would test something like that instead.

We probably should open a tracking issue to track questions like that moving forward.

Reply:
That's not part of the CancellationToken, but of another mechanism. CancellationToken is just for signalling cancellation

Within an integration test we can monitor that the child was cancelled, and that it completed. But we can't really tell whether all internal resources in the `CancellationToken` had been relaesed. The method is more along an `Arc::strong_count()`  method. And since I don't think normal users should be concerned about this I exposed it only for testing purposes and used it there.

Reply:
How about we punt for now as the impl will change soon. We can add more of such tests after?